File: src/templates/scripts/transmitter.py
#!/usr/bin/env python3
import rospy
from std_msgs.msg import String

class Transmitter:

    def __init__(self):
        # rospy.set_param(param_name='hz', param_value=1.0)
        hz = rospy.get_param(param_name='hz')

        self.pub = rospy.Publisher(name='/example_topic', data_class=String, queue_size=10)

        self.publish(hz=hz)
    
    
    def publish(self, hz):
        rate = rospy.Rate(hz=hz)

        while not rospy.is_shutdown():
            msg = String()
            msg.data = 'music'
            self.pub.publish(msg)

            rospy.loginfo(msg=f'Transmitting {msg.data}')

            rate.sleep()

if __name__ == '__main__':
    rospy.init_node(name='transmitter', anonymous=False)
    Transmitter()

File: src/templates/src/transmitter.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

class Transmitter
{
    public:

    Transmitter()
    {
        // nh.setParam("hz", 1.0);
        nh.getParam("hz", hz);
        pub = nh.advertise<std_msgs::String>("/example_topic", 10);
        publish(hz);
    }

    private:

    void publish(double)
    {
        ros::Rate rate(hz);

        while (ros::ok())
        {
            std_msgs::String msg;
            msg.data = "news";
            pub.publish(msg);

            ROS_INFO("Transmitting %s", msg.data.c_str());

            rate.sleep();
        }
    }

    ros::NodeHandle nh;
    ros::Publisher pub;
    double hz;
};

int main(int argc, char **argv)
{
    ros::init(argc, argv, "transmitter", ros::init_options::AnonymousName);
    Transmitter transmitter;

    return 0;
}